### PR TITLE
[CMD] Batch: Don't ClearBatch if bSameFn

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Recursing batch was prevented by `ClearBatch` call.
JIRA issue: [CORE-18093](https://jira.reactos.org/browse/CORE-18093)

## Proposed changes

- Check `bSameFn` variable.
- If it was `TRUE`, don't call `ClearBatch` function.

## TODO

- [x] Do tests.

## Comparison

Win2k3:
Win2k3
Successful.

ReactOS (BEFORE):
before
FAILED.

ReactOS (AFTER):
after
Successful.

### CODE REVIEW

**Comment by HBelusca** [comment]
Could you check whether your patch handles the following scenario?
Two command files named `lol`, but: the first one is `lol.bat` and the second one is `lol.cmd`.
The first file: `lol.bat`, runs a different command than "lol" (in the example below it tries to run the non-existing `ftw` command).
The second file: `lol.cmd`, tries to recursively run itself by calling `lol` (_**without extension**_).
```
X:\temp>dir lol*
 Le volume dans le lecteur X s'appelle Data
 Le numéro de série du volume est 9E4F-2472

 Répertoire de X:\temp

09/07/2024  16:11                 5 lol.bat
09/07/2024  16:11                 5 lol.cmd
               2 fichier(s)               10 octets
               0 Rép(s)  535 165 153 280 octets libres

X:\temp>type lol.bat
ftw

X:\temp>type lol.cmd
lol

REM
REM Try to run "lol.bat" from the prompt.
REM

X:\temp>lol.bat

X:\temp>ftw
'ftw' n'est pas reconnu en tant que commande interne
ou externe, un programme exécutable ou un fichier de commandes.

REM
REM Try to run "lol.cmd" from the prompt.
REM

X:\temp>lol.cmd

X:\temp>lol

X:\temp>ftw
'ftw' n'est pas reconnu en tant que commande interne
ou externe, un programme exécutable ou un fichier de commandes.

REM
REM Now, try to run "lol" without extension, directly from the prompt. It will run lol.bat.
REM

X:\temp>lol

X:\temp>ftw
'ftw' n'est pas reconnu en tant que commande interne
ou externe, un programme exécutable ou un fichier de commandes.

X:\temp>
```

File: base/shell/cmd/batch.c [hunk 1]
```diff
@@ -408,17 +409,9 @@ INT Batch(LPTSTR fullname, LPTSTR firstword, LPTSTR param, PARSED_COMMAND *Cmd)
...
-        BatType = BAT_TYPE;
-
-        /* If this is a .CMD file, adjust the type */
-        dot = _tcsrchr(bc->BatchFilePath, _T('.'));
-        if (dot && (!_tcsicmp(dot, _T(".cmd"))))
-        {
-            BatType = CMD_TYPE;
-        }
+        /* Is the top-level batch context type .CMD or .BAT? */
+        TCHAR *dotext = _tcsrchr(bc->BatchFilePath, _T('.'));
```
Comment (HBelusca) [inline base/shell/cmd/batch.c:413]:
```suggestion
        /* Default the top-level batch context type
         * to .BAT, unless this is a .CMD file */
        PTCHAR dotext = _tcsrchr(bc->BatchFilePath, _T('.'));
```
Besides the NT-ification of the pointer to TCHAR made here,
I suggest the comment change to emphasize more (as was the intent in the original code) the fact that the batch type _always default_ to BAT; that is, unless the file is .CMD.

**Comment by katahiromz** [comment]
The test results.

ReactOS (AFTER):
results

**Review by HBelusca — APPROVED** [review]
Tested it, LGTM :+1:

**Pull request merged**